File: apps/ebill_server/priv/templates/helpers/ruby/ebill.rb
require 'json'

module EBill
  VERSION="0.0.1"

  class <<self
    def info(message) 
      ErlPort::Erlang.call(:error_logger, :info_msg, [message])
    end
    def error(message) 
      ErlPort::Erlang.call(:error_logger, :error_msg, [message])
    end
    def warning(message) 
      ErlPort::Erlang.call(:error_logger, :warning_msg, [message])
    end

    def to_json(data)
      begin
        {:ok => JSON.parse(data)}
      rescue JSON::ParserError => e
        {:error => "JSON Error: #{e}"}
      rescue => e
        {:error => "Internal error: #{e}"}
      end
    end

    def ok(data)
      JSON.generate({:ok => data})
    end

    def error(data)
      JSON.generate({:error => data})
    end
  end
end

File: apps/ebill_server/priv/templates/helpers/python/ebill.py
from erlport.erlterms import Atom
from erlport.erlang import call
import json
import sys


VERSION = "0.0.2"


def info(message):
    call(Atom("error_logger"), Atom("info_msg"), [message])


def error(message):
    call(Atom("error_logger"), Atom("error_msg"), [message])


def warning(message):
    call(Atom("error_logger"), Atom("warning_msg"), [message])


def to_json(data):
    try:
        rc = json.loads(data)
    except ValueError as e:
        rc = {"error": "JSON error #{0}: {1}".format(e.errno, e.strerror)}
    except:
        rc = {"error": "Internal error: {0}".format(sys.exc_info()[0])}
    return rc


def ok(data):
    return json.dumps({"ok": data})


def err(data):
    return json.dumps({"error": data})
